e2e/validation.go

```diff
@@ -10,12 +10,61 @@ import (
 	"github.com/Azure/agentbakere2e/scenario"
 )
 
-func runLiveVMValidators(ctx context.Context, t *testing.T, vmssName, sshPrivateKey string, opts *scenarioRunOpts) error {
-	privateIP, err := getVMPrivateIPAddress(ctx, opts.cloud, opts.suiteConfig.subscription, *opts.chosenCluster.Properties.NodeResourceGroup, vmssName)
+func validateNodeHealth(ctx context.Context, t *testing.T, kube *kubeclient, vmssName string) (string, error) {
+	nodeName, err := waitUntilNodeReady(ctx, kube, vmssName)
 	if err != nil {
-		return fmt.Errorf("unable to get private IP address of VM on VMSS %q: %w", vmssName, err)
+		return "", fmt.Errorf("error waiting for node ready: %w", err)
 	}
 
+	nginxPodName, err := ensureTestNginxPod(ctx, kube, nodeName)
+	if err != nil {
+		return "", fmt.Errorf("error waiting for pod ready: %w", err)
+	}
+
+	err = waitUntilPodDeleted(ctx, kube, nginxPodName)
+	if err != nil {
+		return "", fmt.Errorf("error waiting for pod deletion: %w", err)
+	}
+
+	return nodeName, nil
+}
+
+func validateWasm(ctx context.Context, kube *kubeclient, nodeName, privateKey, vmPrivateIP string) error {
+	spinPodName, err := ensureWasmPods(ctx, kube, nodeName)
+	if err != nil {
+		return fmt.Errorf("failed to valiate wasm, unable to ensure wasm pods on node %q: %w", nodeName, err)
+	}
+
+	spinPodIP, err := getPodIP(ctx, kube, defaultNamespace, spinPodName)
+	if err != nil {
+		return fmt.Errorf("unable to get pod IP of wasm spin pod %q: %w", spinPodName, err)
+	}
+
+	debugPodName, err := getDebugPodName(kube)
+	if err != nil {
+		return fmt.Errorf("unable to get debug pod name to validate wasm: %w", err)
+	}
+
+	curlCmd := fmt.Sprintf("curl http://%s/hello", spinPodIP)
+
+	execResult, err := pollExecOnPod(ctx, kube, defaultNamespace, debugPodName, curlCmd)
+	if err != nil {
+		return fmt.Errorf("unable to execute wasm validation command: %w", err)
+	}
+
```

Comment:
maybe add another "retry get pod ip + curl spin" step here if the first execResult has exit code 7 (connection failure)